datalad/interface/annotate_paths.py

```diff
@@ -560,6 +576,10 @@ def __call__(
                 report_untracked='all' if force_untracked_discovery else 'no',
                 recursion_limit=recursion_limit)
 
+            from itertools import chain
+            # re-append the preserved paths:
+            requested_paths = chain(requested_paths, iter(preserved_paths))
+
```

Comment:
Aren't these two plain (non-nested) lists? If so, why not just `requested_paths.extend(preserved_paths)`?

Reply:
Nope. After modification detection `requested_paths` is a generator.

Reply:
I knew there would be something. ACK.